### PR TITLE
Disable a self-notification possibility.

### PR DESCRIPTION
**Pull request by cmrd-senya**
a fix suggestion for #6506.

### CODE REVIEW

File: app/models/notification.rb [hunk 1]
```diff
@@ -15,7 +15,7 @@ def self.for(recipient, opts={})
   end
 
   def self.notify(recipient, target, actor)
-    return nil unless target.respond_to? :notification_type
+    return nil unless target.respond_to? :notification_type || recipient != actor
```
Comment (diaspora-code-review) [inline app/models/notification.rb]: Use parentheses in the method call to avoid confusion about precedence.

Comment (jhass) [inline app/models/notification.rb]: I'm afraid that's not changing anything, you got to retest.

``rb
[50] pry(main)> puts "foo" || "bar"
foo
=> nil
[51] pry(main)> puts("foo") || "bar"
foo
=> "bar"
```

Comment (cmrd-senya) [inline app/models/notification.rb]: Well, that just means it doesn't reproduce for me. I had only one reproduction and then everything worked fine every time.

Is it fine if I do parenthesis:
```ruby
-    return nil unless target.respond_to? :notification_type || recipient != actor
+    return nil unless (target.respond_to?(:notification_type) || recipient != actor)
```

Comment (jhass) [inline app/models/notification.rb]: It would be very good to have a test that fails without the change :/

Comment (cmrd-senya) [inline app/models/notification.rb]: This new tests don't fail for me without a change.

@AugierLe42e, can you please help us reproducing the issue?

Comment (AugierLe42e) [inline app/models/notification.rb]: What the f? I didn't get that mention. The power of zoning randomly. What do you need exactly, @cmrd-senya ?

Comment (cmrd-senya) [inline app/models/notification.rb]: Not only we who have notification problems :)

Well, you can check if this tests fail in your setup. For me #6506
doesn't reproduce, so I need some steps to reproduce it.

Comment (AugierLe42e) [inline app/models/notification.rb]: You mean you want me to merge this PR on my pod and see if solves #6506?

Comment (jhass) [inline app/models/notification.rb]: @AugierLe42e no, we'd like a failing spec for the issue.

Comment (cmrd-senya) [inline app/models/notification.rb]: @AugierLe42e, are you able to certainly reproduce it?

Comment (AugierLe42e) [inline app/models/notification.rb]: Really, I don't understand what you are waiting for me. You want me to run these tests on my computer ?

Comment (Flaburgan) [inline app/models/notification.rb]: @cmrd-senya is not able to reproduce the bug, @AugierLe42e you're the one who reported it so he's asking for a way to reproduce it. The perfect way to give the steps to reproduce would be a test. But you can also describe it here.

Comment (cmrd-senya) [inline app/models/notification.rb]: Yes, that what I was going to answer :)

I believe the issue exists, I think I saw it once myself, but never
managed to reproduce.

Comment (AugierLe42e) [inline app/models/notification.rb]: Just reshare any public post and comment it. You should see a notification saying that you commented your own reshare. Reproduced on diaspora-fr.org. diaspote is not up to date.

Comment (AugierLe42e) [inline app/models/notification.rb]: It also reproduces on original posts but not all of them.

Comment (cmrd-senya) [inline app/models/notification.rb]: I don't have an account on diaspora-fr, and if I had, I doubt @Flaburgan
would let me tamper with the running pod to debug the issue.

I need a way to reproduce it on a fresh Diaspora installation.

Comment (Flaburgan) [inline app/models/notification.rb]: Ping me on IRC if you need an account, no problem with that.

@AugierLe42e the "comment on my own reshare" is tested in this PR. As you said, it's not the only case.

Comment (AugierLe42e) [inline app/models/notification.rb]: Yep. It also does it on some regular posts. No pattern found yet.

**Pull request closed without merging**